src/settings/lib/RuleEditor/rules-show-hint.js

```diff
@@ -709,6 +728,56 @@ class MultipleSelectionHintSection extends HintSection {
 
     this.container.classList.add('CodeMirror-hints-multiple-selection');
     this.initCompletionButton(sectionOptions.buttonText);
+
+    if (!this.header) {
+      this.initHeader('');
+    }
+
+    this.initFilterField(sectionOptions.filterPlaceholder);
+  }
+
+  initFilterField(filterPlaceholder) {
+    this.filterInput = document.createElement('input');
+    this.filterInput.type = 'text';
+    this.filterInput.placeholder = filterPlaceholder;
+    this.filterInput.id = 'filter-locations-input';
+    this.filterInput.oninput = this.handleFilterInputChange;
+
+    this.header.appendChild(this.filterInput);
+  }
+
+  handleFilterInputChange = () => {
+    if (isEmpty(this.itemsOptions)) return;
+
+    this.clearItemsActivation();
+    this.filterItems(this.filterInput.value);
+    this.changeActive(this.defaultSelectedHintIndex);
+  }
+
+  clearItemsActivation() {
+    super.getListNodesArray().forEach(listNode => {
+      listNode.classList.toggle(ACTIVE_HINT_ELEMENT_CLASS, false);
+    });
+  }
+
+  filterItems(filterValue) {
+    if (isEmpty(this.itemsOptions)) return;
+
+    const filteredItems = this.itemsOptions.filter(itemOptions => {
+      return itemOptions.displayText.toLowerCase().includes(filterValue.toLowerCase());
+    });
+
+    const itemsIdsToDisplay = filteredItems.map(itemOptions => itemOptions.id);
+
+    super.getListNodesArray().forEach((currentListItem) => {
```

Comment:
for consistency with a `listNodes` naming
```suggestion
    super.getListNodesArray().forEach((listNode) => {
```